src/fields/multi_field.impl.hh

```diff
@@ -158,37 +155,66 @@ template<int spacedim, class Value>
 void MultiField<spacedim,Value>::set_input_list(const Input::Array &list) {
     if (! flags().match(FieldFlag::declare_input)) return;
 
-	set_up_components();
-	for( SubFieldType &field : sub_fields_) {
-		field.set_input_list(list);
-	}
+    // Check sizes of Arrays defined MultiField in field descriptors
+    for (Input::Iterator<Input::Record> it = list.begin<Input::Record>();
+					it != list.end();
+					++it) {
+    	Input::Array mf_array;
+    	if ( it->opt_val(this->input_name(), mf_array) ) {
+    		unsigned int comp_size = this->shared_->comp_names_.size();
+    		if (mf_array.size() != 1 && mf_array.size() != comp_size)
+    			THROW( Input::ExcInputMessage() << EI_Message("Invalid size of Array defined for MultiField " + this->input_name()
```

Comment:
Define new type of the input exception.
The message should contain actual and expected size, e.g.
'Invalid size XX of the MultiField <name>, expected size: YY. '
No need to add address explicitly into the message, see INPUT_EXCEPTION usage.